src/frontend/renderer.nim

```diff
@@ -488,6 +488,12 @@ proc onContextStartHistory*(sender: js, response: jsobject(inState=bool, express
 proc onLoadParsedExprsReceived*(sender: js, response: jsobject(argId=cstring, value=JsAssoc[cstring, seq[FlowExpression]])) =
   jsAsFunction[proc(response: JsAssoc[cstring, seq[FlowExpression]]): void](data.network.futures["load-parsed-exprs"][response.argId])(response.value)
 
+proc onUploadedTraceReceived*(sender: js, response: jsobject(argId=cstring, value=UploadedTraceData)) =
```

Comment:
those should have the exact same names: e.g. `onUploadTraceFile` (or `futures["uploaded-trace"]` ): sorry for not autogenerating them currently

Reply:
probably should rename ipc message from index to `upload-trace-file-received`: that's what we currently do for `await indexCode()`: sending `index-code-received` back